dvc/remote/s3.py

```diff
@@ -325,12 +325,15 @@ def _upload(self, from_file, to_info, name=None, no_progress_bar=False):
             )
 
     def _download(self, from_info, to_file, name=None, no_progress_bar=False):
-        if no_progress_bar:
-            total = None
-        else:
-            total = self.s3.head_object(
-                Bucket=from_info.bucket, Key=from_info.path
-            )["ContentLength"]
+        total = self.s3.head_object(
+            Bucket=from_info.bucket, Key=from_info.path
+        ).get("ContentLength", 0)
+        if total == 0:
+            basedir = os.path.dirname(to_file)
+            os.makedirs(basedir, exist_ok=True)
+            open(to_file, "w").close()
+            return
```

Comment:
I don't think this one is needed, as it seems to work fine with regular s3.

Reply:
Sadly it wont work without this return (and creating file is also needed), because similar error is raised in boto3:

```
  File "venv/lib/python3.6/site-packages/s3transfer/download.py", line 346, in _submit
    response['ContentLength'])
KeyError: 'ContentLength'
```
So my solution basically doesn't run `self.s3.download_file` if the file is empty to overcome this issue.

Reply:
Ok sorry, I misunderstood your comment. Are you absolutely positive that it works with regular s3? If so there must be a 'ContentLength' set to zero there.

Reply:
Yes, it works with s3 because it sets ContentLength to 0. Ok, so this is even worser than we thought. If s3transfer also doesn't work with it, then it is an even bigger problem for minio. Could you please report this issue to minio guys? Let's see what they have to say about it.

Reply:
Sure, we could use this hack in dvc just for the minio for now, but ideally, we need to fix it in minio or in boto3/s3transfer as well.

Reply:
Ok, so I will make sure its is not my minio version problem (note my comment in the issue) and then (if necessary) I will report this problem to minio guys :) Keep in touch